python/pyarrow/filesystem.py

```diff
@@ -237,12 +238,28 @@ class LocalFileSystem(FileSystem):
 
     _instance = None
 
+    def __init__(self):
+        warnings.warn(
+            "pyarrow.filesystem.LocalFileSystem is deprecated as of 2.0.0, "
+            "please use pyarrow.fs.LocalFileSystem instead",
+            DeprecationWarning, stacklevel=2)
```

Comment:
Put a helper in `pyarrow.util`? (and use `FutureWarning`?)

Reply:
There is a `_deprecate_class` in ``pyarrow.util``, but that serves a different purpose (it aliases the old name to the new class, but with warning). 

And if it is just a helper for raising the warning, I am not sure that is worth it (compared to using a template message that can be filled in).

Reply:
Moved a template to `pyarrow.util`. Can also make it a function raising the warning if you prefer